### ISSUE TITLE
Overlay for multilingual names

### ISSUE THREAD

**Issue by arrival-spring** [streetcomplete/StreetComplete#4876]
Affected tag(s) to be modified/added: [name:**](https://wiki.openstreetmap.org/wiki/Names#Language-specific_names)
Question asked: **What are the names of this object?**

Linked: #4220 and #3411

There should be a setting when/before selecting the overlay for which languages it is relating to. e.g. I could select "French" and "German".

The overlay would highlight as incomplete any object which only has `name` and neither `name:fr` nor `name:de`

The overlay would use one colour for objects with `name` and `name:fr` but without `name:de`. Likewise, a different colour for when German is tagged but not French.

Highlight as complete objects with `name`, `name:fr` and `name:de`

I can tap on any object with a name and something like the name quest form comes up where I can add and change names in these languages (if any other languages was mapped as well then that would also be shown and editable).

There needs to be a way to say what the main name is (`name`). In some areas the convention is that this is not the same as any of the sublanguage tags (e.g. road names where multiple languages are combined with "/"). In most other cases it is identical to one of the sublanguage names. This could be a dropdown to select between the languages already typed or "something else...".

Something like this:
* Main name: (dropdown with options of the other languages, or "something else..." leading to a free text box
* fr: Free text box
* de: Free text box

**Comment by rhhsm** [comment]
This overlay would be very useful for areas where scripts different from the Latin script are used. In Bulgaria, for instance, the convention seems to be that any name should be added 3 times: `name` (usually the Bulgarian name, in Cyrillic script), `name:bg` (to ensure that the language of the name is verified) and `int_name` (for the transliteration of the name in Latin script). Often, instead of `int_name`, `name:en` is given because this is easier to add in iD (`int_name` is not listed among the language options in the **Multilingual Name** drop-down list). If no name in Latin script is available, renderers often use the transliteration for Russian Cyrillic to show the name in Latin script, but this is not consistent with transliteration for Bulgarian Cyrillic so differences with actual signposts in the field will occur (Хасково should be Haskovo, not Khaskovo). I.e. for Bulgaria, SC should show the `int_name` box prominently (which it already does for the name quest).

**Comment by arrival-spring** [comment]
I guess when you have the setting for the overlay of which languages to choose it could default to the language list for the country (as used by name quests)

**Comment by westnordost** [comment]
Also, couldn't the recording of such incomplete data not be done with quests? After all, there is usually little reason to later change the name(s).

**Comment by westnordost** [comment]
On the other hand, would need a way to mark a street (etc) as not having multiple names

**Comment by arrival-spring** [comment]
> We have data for this:
Ah yes, that makes sense. Although there are some countries with 6 or more languages listed there, which could make it a challenge to show all of them and/or having different colours depending on which of them is missing.

The argument for using an overlay (rather than a quest) is that in many multilingual areas, not all objects have names in every language and so it is spammy to ask it for all elements. That was the main reason for closing #4220

Additionally, there was the discussion in #3411 of situations where `name` is 'wrong', that is it is not in the local language and should be in `name:en` or something, and a local name in `name:**`, but there was no way to identify such objects (we can't really write something to detect the language of the name tag).

**Comment by westnordost** [comment]
Yeah, I was about to ref to #3411. For certain countries, `name` must always be in a certain script. I.e. in Thailand in Thai script, in Russia in Cryllic, in Greece in Greek, in Georgia in Georgian script, in Myanmar in Burmese script etc.. So that would be quite a nice and unproblematic quest. One would just need to collect in which countries which scripts are used.

**Comment by arrival-spring** [comment]
I've replied there with an issue about that.

But the idea behind this overlay is for other countries with incomplete multilingual tagging (but where such tagging is desired) where script detection would not catch such cases

**Comment by rhhsm** [comment]
I think this job can be done with a quest that just has a slightly different icon design than the corresponding one for items without any name (different background colour?). However, it should be considered that:

1. Most privately owned items such as shops have only one name, so I don't think we should apply this to the place name quest. It should only be applied to public names such as street names, public transport stop names and other geographic names.
2. The data for which languages to ask for should be verified by asking the various country OSM communities (maybe they already are?)
3. Which languages to ask for may vary within one country (India has 22 different official languages using 12 different scripts).
4. The names already tagged should be displayed and be editable, so "wrong" names can be corrected and names can be copy-pasted if they are the same in the different languages asked for.
5. Although an item may have several official names in different languages and scripts, this may not always be on the sign. Could we encourage users to enter those names if they know them, even if they are not signed? For Bulgaria, for instance, street name signs are not regularly renewed, but there is an official rule on how names should be transliterated from Cyrillic to Latin script. Could we add instructions on how to find the name in another script/language? Maybe this could be done through targeted translation of instructions in POEditor, i.e. the Bulgarian language version of SC tells the Bulgarian user how to transliterate, etc.? (wondering how to do this for Switzerland without bothering German/French/Italian users...)

**Comment by matkoniecz** [comment]
> I can tap on any object with a name and something like the name quest form comes up where I can add and change names in these languages

Note that for example in Poland vast majority of objects with `name` has no English, German or Chinese names.

**Comment by arrival-spring** [comment]
> Note that for example in Poland vast majority of objects with `name` has no English, German or Chinese names.

But English, German or Chinese would only be showing if you'd selected them to, or if they were in `officialLanguages.yml` or `additionalStreetsignLanguages.yml` which they are not for Poland

**Comment by rhhsm** [comment]
@matkoniecz We should indeed also take monolingual mono-script countries into account and see that such a quest/overlay should do for them. According to the data quoted by @westnordost , in PL the only official language is pl. I see that it's recommended to repeat any name tag with the corresponding language specific tag [](https://wiki.openstreetmap.org/wiki/Multilingual_names#Repeating_name_with_language_specific_tag) but this seems to make sense to me only when there are name tags for other languages than pl. So the quest could ignore items than only have a `name` tag (and assume it's in Polish) and only look for named items that have a tag for another language than Polish but not a `name:pl` tag, and ask for it. It's very likely that it will be the same value as the `name` value, so a button could be added that copies this value into the `name:pl` field. Would that be useful behaviour for the Polish case?

**Comment by matkoniecz** [comment]
I see that it's recommended to repeat any name tag with the corresponding language specific tag 

Note

> In cases where multiple languages are tagged for a name

I added https://wiki.openstreetmap.org/wiki/Multilingual_names#When_name_is_only_in_one_language

**Comment by westnordost** [comment]
> Which languages to ask for may vary within one country (India has 22 different official languages using 12 different scripts).

We got that data:

https://github.com/streetcomplete/countrymetadata/blob/6d136241fb0b6ce47b08d8288e74c6ca3cdbd62a/data/officialLanguages.yml#L169-L206

**Comment by rhhsm** [comment]
I agree: it's important data that should be added, but SC is probably not the most suitable tool to add them because a survey is not needed and they can be added by an armchair mapper. It would be more suitable to have an iD validator tool for this.
